### ISSUE TITLE
A11y: BlockSection toggle-display="switch". Can't use keyboard.

### ISSUE THREAD

**Issue by asangma** [Esri/calcite-app-components#801]
### Summary
When a block-section has `toggle-display="switch"`, the label, which triggers showing/hiding of the content, can't be focused. The switch can be focused, but keyboard commands don't toggle the content.

### Actual Behavior
![block-section-focus2](https://user-images.githubusercontent.com/12503298/73676809-de1a5700-4669-11ea-8ded-bbde57643226.gif)

### Expected Behavior
* ability to focus the label node
* keyboard accessibility to toggle section content when `toggle-display="switch"`

### Reproduction Steps

1. See block demo
1. Add `toggle-display="switch"` to one of the sections.

### Relevant Info 
Mentioned in #527.

cc @kat10140

**Comment by asangma** [comment]
That cat is all like, _SUCH A11Y. MUCH INTRIGUE._

**Comment by jcfranco** [comment]
Installed. 

**Note**: The switch will still be keyboard-focusable until we update calcite-components to v1.0.0-beta.19 (Esri/calcite-components#317)

**Comment by asangma** [comment]
Très bon!

**Comment by asangma** [comment]
@AdelheidF Forgot to tag you on this. The block-section switch mode should be fixed.

**Comment by AdelheidF** [comment]
The toggle itself can no loger get the focus. I assume this is the fix. So all good now.

![image](https://user-images.githubusercontent.com/44481292/75077165-d276bf00-54b6-11ea-9b9f-d489a9bb3377.png)

**Comment by jcfranco** [comment]
@AdelheidF Correct. 😄

**Comment by AdelheidF** [comment]
Just as an FYI, this makes it weird in certain cases.
E.g. here the first and third toggle buttons are part of sections, the middle one is just a toggle button by itself.

Flow of focus:
![image](https://user-images.githubusercontent.com/44481292/75177337-cd01ba80-56ea-11ea-9152-489f1dd2b3a3.png)

![image](https://user-images.githubusercontent.com/44481292/75177359-d7bc4f80-56ea-11ea-89c5-cd56806e1d55.png)

![image](https://user-images.githubusercontent.com/44481292/75177381-e276e480-56ea-11ea-82e8-8b56c1952b69.png)
